netplan_cli/cli/state.py

```diff
@@ -328,13 +348,22 @@ def vendor(self) -> str:
     @property
     def ssid(self) -> str:
         if self.type == 'wifi':
+            if self.backend == "NetworkManager":
+                return self.query_nm_ssid(self.nm.get('name', ''))
             # XXX: available from networkctl's JSON output as of v250:
             #      https://github.com/systemd/systemd/commit/da7c995
+            # TODO: Retrieving the SSID from systemd seems to not be reliable.
+            #       Sometimes it will return "(null)".
             for line in self._networkctl.splitlines():
                 line = line.strip()
-                key = 'WiFi access point: '
-                if line.startswith(key):
-                    ssid = line[len(key):-len(' (xB:SS:ID:xx:xx:xx)')].strip()
+                key = r'^Wi-?Fi access point: (.*) \(.*\)'
+                if match := re.match(key, line):
+                    ssid = match.group(1)
+                    # TODO: Find a better way to retrieve the SSID
```

Comment:
nitpick (non-blocking): We should try getting this information from the `networkctl --json=short` output instead.

IIRC, this was not implemented back when we first wanted to use it in `netplan status`, but I can see it on systems using systemd v255+ today: `networkctl --json=short status wlp0s20f3 | jq | grep SSID`

Reply:
Sometimes it's not available. The output below is from one of my laptops:

```
danilo@danilo-Aspire-ES1-572:~$ networkctl --json=short status wlp2s0 | jq | grep SSID
  "BSSID": [
danilo@danilo-Aspire-ES1-572:~$
```

On another laptop it will give me the SSID though. I suspect it might be some limitation of the driver, I don't really know.
That's the reason why I changed it to get it from Network Manager when NM is the backend.

Reply:
And in these cases we will show it like this on `netplan status`:

```
●  5: wlan2 wifi/"(null)" UP (networkd: wlan2)
```